src/@carbon-info/views/Ecosystem/Ecosystem.tsx

```diff
@@ -0,0 +1,259 @@
+
+const Ecosystem: React.FC = () => {
+  const classes = useStyles();
+  const { ref, inView } = useInView({
+    /* Optional options */
+    threshold: 0.35,
+    triggerOnce: true,
+  });
+
+  const tabs = ["Featured dApps", "Blockchains", "Wallets", "Validators"];
+  const filters = ["All", "IBC", "EVM", "Non-EVM"]; // TODO: Add Coming Soon filter
+  const [value, setValue] = useState<string>(tabs[0]);
+  const [blockchainFilter, setBlockchainFilter] = useState<string>(filters[0]);
+
+  const handleChange = (newValue: any) => {
+    setValue(newValue);
+  };
+
+  const filterButtons = (
+    <Box className={classes.filters}>
+      {filters.map((filter, index) => {
+        return (
+          <Box className={classes.filterBox} key={`${filter}-${index}`}>
+            <Button className={clsx(classes.filterButton, { active: blockchainFilter === filter })} onClick={() => setBlockchainFilter(filter)} disableRipple>
+              {filter} Chains
+            </Button>
+          </Box>
+        );
+      })}
+    </Box>
+  );
+
+  const filteredBlockchains = React.useMemo(() => {
+    let filtered: BlockchainConfig[] = [];
+    switch (blockchainFilter) {
+      case "IBC":
+        filtered = allBlockchains.filter((blockchain) => {
+          return blockchain.chain === "IBC";
+        });
+        break;
+      case "EVM":
+        filtered = allBlockchains.filter((blockchain) => {
+          return blockchain.chain === "EVM";
+        });
+        break;
+      case "Non-EVM":
+        filtered = allBlockchains.filter((blockchain) => {
+          return blockchain.chain === "Non-EVM";
+        });
+        break;
+      case "All":
+        filtered = allBlockchains;
+        break;
+      default:
+        filtered = allBlockchains;
+        break;
+    }
+    return filtered;
+  }, [blockchainFilter]);
+
+  const sortedValidators = React.useMemo(() => {
+    return allValidators.sort((v1: ValidatorConfig, v2: ValidatorConfig) => {
+      // sorted in descending order
+      return v2.votingPower - v1.votingPower;
+    });
+  }, []);
+
+  const content = React.useMemo(() => {
+    switch (value) {
+      case "Featured dApps":
+        return <AppsCarousel items={allFeatured} inView />;
+      case "Blockchains":
+        return (
+          <>
+            {filterButtons}
+            <div className={classes.contentBox}>
+              <FeatureGrid items={filteredBlockchains} inView />
+            </div>
+          </>
+        );
+      case "Wallets":
+        return (
+          <div className={classes.contentBox}>
+            <FeatureGrid items={allWallets} inView />
+          </div>
+        );
+      case "Validators":
+        return (
+          <div className={classes.contentBox}>
+            <FeatureGrid items={sortedValidators} inView />
+          </div>
+        );
+      default:
+        return <AppsCarousel items={allFeatured} inView />;
+    }
+  }, [inView, value, filteredBlockchains]);
```

Comment:
Try not to render components inside a useMemo as you'll have to track all the dependencies you pass in. Ideally, should do conditional rendering, i.e.
```
{case === "Validators" && (
  <div className={classes.contentBox}>
    <FeatureGrid items={sortedValidators} inView />
  </div>
)}
{case === "Wallets" && (
  ...
)}
```

I'll be refactoring that on Demex soon as well.